src/js/models/menubar.js

```diff
@@ -238,6 +238,7 @@ define(function (require, exports, module) {
             templateMenu = fileMenu.submenu[templateIndex],
             templateSubmenu = templateMenu.submenu;
 
+        templateSubmenu.length = 0;
```

Comment:
It would be less scary to do something like `templateMenu.submenu = templates.map(...)`. Can you try that instead?